lib/appSession.js

```diff
@@ -229,7 +289,13 @@ module.exports = (config) => {
       attachSessionObject(req, sessionName, {});
     }
 
-    onHeaders(res, setCookie.bind(undefined, req, res, { iat }));
+    const { end: origEnd } = res;
+    res.end = async function resEnd(...args) {
+      await store.set(existingSessionValue, req, res, {
+        iat,
+      });
+      origEnd.call(res, ...args);
+    };
```

Comment:
How are storage errors propagated?

Reply:
I'm assuming you'd log them on the store instance (or the client instance that the store is using, eg https://github.com/tj/connect-redis#how-to-log-redis-errors)

I don't _think_ express-sessions has a better solution than this

Reply:
Hmmm... actually, maybe we do need to do something with the `next` https://github.com/expressjs/session/blob/master/index.js#L337

I thought `res.end` was too late to call it but maybe not...